src/transaction/transaction_builders.cpp

```diff
@@ -12,14 +12,17 @@ std::shared_ptr<bytes> stringToBytesPtr(std::string const &in)
 }
 }
 
+// -------------------- EGLD Transfer --------------------
 TransactionEGLDTransferBuilder::TransactionEGLDTransferBuilder(TransactionBuilderInput txInput) :
         ITransactionBuilder(),
         m_txInput(std::move(txInput))
 {}
 
 Transaction TransactionEGLDTransferBuilder::build()
 {
-    uint64_t gasLimit = m_txInput.gasEstimator.forEGLDTransfer(m_txInput.data.size());
+    uint64_t inputGasLimit = m_txInput.gasLimit;
+    uint64_t estimatedGasLimit = m_txInput.gasEstimator.forEGLDTransfer(m_txInput.data.size());
+    uint64_t gasLimit = (inputGasLimit == DEFAULT_GAS_LIMIT) ? estimatedGasLimit : inputGasLimit;
```

Comment:
Somehow, if the user provides a gas limit equal to DEFAULT_GAS_LIMIT, that's ignored. However, the result is still good :)

So, tricky, but correct - can you double check, though?

Reply:
User cannot provide gasLimit, only gasPrice. GasLimit is always set by the factory in builders